net/dns/resolver/tsdns.go

```diff
@@ -725,6 +725,12 @@ func (r *Resolver) resolveLocalReverse(name dnsname.FQDN) (dnsname.FQDN, dns.RCo
 		return "", dns.RCodeRefused
 	}
 
+	// If the requested IP is part of the IPv6 4-to-6 range, it actually
+	// corresponds to a IPv4 address.
+	if ip.Is6() && tsaddr.Tailscale4To6Range().Contains(ip) {
+		ip = tsaddr.Tailscale6to4(ip)
+	}
```

Comment:
but what if they have IPv4 disabled on their tailnet? won't `r.ipToHost` then be keyed off IPv6?

maybe it should just have both address family map keys?

Reply:
TIL we have that option :smile: :sweat_smile:

Looking through control, yes - seems `populateNodeIPs` removes all v4 addresses if the `DisableIPv4` bit is set. I'll go ahead and update the logic to do the map lookup on both the v6 and v4 variant.